syndiffix/bucket.py

```diff
@@ -223,23 +225,26 @@ def _harvest_branch(branch: Branch, harvested_nodes: BucketsCache) -> Buckets:
         if branch.dimensions() == 1:
             children_buckets = [Bucket(tuple(branch.bucket_intervals()), parent_count)]
         else:
-            children_buckets += _refine_buckets(branch, harvested_nodes, parent_count - children_count)
+            children_buckets += _refine_buckets(branch, harvested_nodes, parent_count - children_count, unsafe_rng)
     else:
         _adjust_counts(children_buckets, children_count, parent_count)
 
     return children_buckets
 
 
-def _harvest_node(node: Node, harvested_nodes: BucketsCache) -> Buckets:
+def _harvest_node(node: Node, harvested_nodes: BucketsCache, unsafe_rng: Random) -> Buckets:
     if harvested_nodes.get(node) is None:
         harvester = _harvest_leaf if isinstance(node, Leaf) else _harvest_branch
-        harvested_nodes[node] = harvester(node, harvested_nodes)  # type: ignore
+        harvested_nodes[node] = harvester(node, harvested_nodes, unsafe_rng)  # type: ignore
 
     return harvested_nodes[node]
 
 
 def harvest(node: Node) -> Buckets:
     # Maps node.NodeData.Id to its already harvested buckets, so it's done only once.
     harvested_nodes: BucketsCache = dict()
-    buckets = _harvest_node(node, harvested_nodes)
+    # This RNG isn't safe for anonymization, so it can only be applied to already anonymized data.
+    # Used for randomly matching subintervals during refining.
+    unsafe_rng = Random(0)
```

Comment:
I feel like this might produce a bias. Can we derive an rng from `node.context`?